### ISSUE TITLE
Chapter 28 -  Curve-line intersections

### ISSUE THREAD

**Issue by janinurminen** [Pomax/BezierInfo-2#329]
By selecting control points for quadratic bezier carefully you can see that intersection code doesn't handle border cases properly.

This happens e.g. when you have following setup where p0(180, 0), p1(180, 100) and p2(180, 200), line in y=100:

<img width="302" alt="Screenshot 2021-10-09 at 14 32 56" src="https://user-images.githubusercontent.com/5229523/136656338-ef48eefe-0634-4956-8234-ab2acf65f8ad.png">

If you move horizontal line up/down you will discover that there are also other spots where intersection fails.

**Comment by Pomax** [comment]
Yeah, degenerate curve cases definitely need some fixing atm. (You'd think the math worked whether curves were degenerate or not, but no. Zero-deltas make that hilariously not-applicable)
